### DISCUSSION TITLE
Relative paths

### DISCUSSION THREAD

**Discussion by damiankomonski** [webpack/webpack#17290] [Q&A]
Hello!

I have problem with paths in webpack. Webpack bundle everything and always return absolute paths. Piece of my webpack configuration:

```
            {
                test: /\.(png|svg|jpg|jpeg|gif)/i,
                type: "asset/resource",
                generator: {
                    outputPath: "../images/",
                    publicPath: "wp-content/themes/szablon/assets/images/"
                }
            },
            {
                test: /\.(ttf|otf|woff|woff2)/i,
                type: "asset/resource",
                generator: {
                    outputPath: "../fonts/",
                    publicPath: "wp-content/themes/szablon/assets/fonts/"
                }
            }
```

I have to write such paths in publicPath. What I want is simple relative path:
`./szablon/`
        

If I write this in publicPath, webpack bundle and throws out absolute path:
`/szablon/`
        

Please help, how can I do this in webpack. I am using version 5.78.0.

**Comment by alexander-akait** [comment]
webpack knows nothing about your structure, so if you write `wp-content/themes/szablon/assets/fonts/` you will have it, if you want relative paths, don't touch `publicPath` or set it `publicPath: "auto"`

**Reply by damiankomonski** [reply]
I set it and I still get paths like
`{background:url(about-us-photo-2.png) no-repeat;`

**Reply by damiankomonski** [reply]
Exactly what I want achieve
`./../images/about-us-photo-2.png`

**Reply by alexander-akait** [reply]
But `about-us-photo-2.png` is relative URL, webpack undestand where file was located and how simplify path, you can define publicPath manually in mini-css-extract-plugin, but I don't recommend it

**Reply by damiankomonski** [reply]
I try to set it in mini-css-extract-plugin in this way

```
            {
                test: /\.(scss)$/,
                use: [
                    {
                        loader: MiniCssExtractPlugin.loader,
                        options: {
                            publicPath: "./../"
                        }
                    },
                    "css-loader",
                    {
                        loader: "postcss-loader",
                        options: {
                            postcssOptions: {
                                plugins: [
                                    require("autoprefixer"),
                                    require("cssnano")({ preset: "default" })
                                ]
                            }
                        }
                    },
                    "sass-loader"
                ]
            },
```

and still I get such code
`background:url(/images/icon-arrow-right-blue2.svg) no-repeat;`

**Reply by alexander-akait** [reply]
Can you tell me why you are trying to achieve this behavior, I get the feeling that you try to solve something bad using hacky ways

**Reply by damiankomonski** [reply]
I am working locally, if I compile my project it runs, but if I want deploy it on server paths not always match. I have to change them every time. It is a little annoying. If I had relative paths, they would always match and be correct.

Am I doing something wrong? I remember that in Gulp for example I could do this. Hmm....

**Reply by damiankomonski** [reply]
It seems to be simple thing, but maybe I don't understand something.

**Reply by alexander-akait** [reply]
Can you try `/wp-content/themes/szablon/assets/fonts/`, i.e. starting with `/`

**Reply by damiankomonski** [reply]
It doesn't work. My project is in subfolder, I need path like:
"./something/something"

**Reply by alexander-akait** [reply]
In this case please create reproducible test repo using github, sorry I can't undestand your structure

**Reply by alexander-akait** [reply]
And I strongly recommend do not touch public path, by default is relative urls, and when you try to change something you need to change other things and after this you can just copy/paste or change location of files without pain
